app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt

```diff
@@ -97,25 +98,19 @@ class HomeMenu(
         }
     }
 
-    private fun syncSignInMenuItem(): BrowserMenuImageText? {
-        val syncItemTitle =
-            if (context.components.backgroundServices.accountManagerAvailableQueue.isReady()) {
-                accountManager.accountProfileEmail ?: context.getString(R.string.sync_menu_sync_and_save_data)
-            } else {
-                null
-            }
```

Comment:
Looks to me like this is the only difference - show "Syncing.." instead of not showing a menu option at all?
In which case the new text could be added here instead of `null` without the need of a new `BrowserMenuSignIn` class.

Reply:
If we add text here, then the item will be build with the "syncing..." text and won't be change on the consecutive menu calls, until the user leaves the screen and the item gets recreated. 

The main difference is that `BrowserMenuSignIn` uses a function to get text every time instead of receiving it as a parameter. Calling the function during the `bind`allows `BrowserMenuSignIn` to get the most recent state from the account manager.

I will discuss the phrasing with UX